src/V3LinkLValue.cpp

```diff
@@ -205,6 +205,17 @@ class LinkLValueVisitor : public AstNVisitor {
         }
         m_setRefLvalue = last_setRefLvalue;
     }
+    void prepost_visit(AstNodeTriop* nodep) {
+        m_setRefLvalue = true;
+        iterateAndNextNull(nodep->lhsp());
+        m_setRefLvalue = false;
+        iterateAndNextNull(nodep->rhsp());
+        iterateAndNextNull(nodep->thsp());
```

Comment:
Ideally the thsp should be the lvalue and shuffle how you use them appropriately.  This is because the "processing" order of the statements in op order will then be closer to sequential access that is op1: constant, op2 read old value, op3. set new value.

Reply:
I am not sure what you mean by "shuffle how you use them". Do you mean that I should just set thsp to lvalue too? Could you please elaborate on what exactly is needed here?